src/DataObject/GridColumnConfig/Operator/Substring.php

```diff
@@ -84,13 +91,35 @@ public function getLabeledValue(array|ElementInterface $element): ResultContaine
             if ($isArrayType) {
                 $result->value = $valueArray;
             } else {
-                $result->value = $valueArray[0];
+                $result->value = $valueArray[0] ?? '';
             }
         }
 
         return $result;
     }
 
+    private function convertToString(mixed $value): string
+    {
+        if (is_array($value)) {
+            $output = implode(
+                ' ',
+                array_map(
+                    fn($v) => is_scalar($v) ? (string)$v : '',
+                    array_values($value)
```

Comment:
This is unnecessary, as `implode` does not care about array keys.

```suggestion
                    $value
```